File: 38.count-and-say/solution.cpp
class Solution {
public:
	string countAndSay(int n) {
		string s = "1";
		for (int i = 1; i < n; i++)
		{
			stringstream ss;
			char last = '.';
			int cnt = 0;
			for (char c : s)
			{
				if (last != c)
				{
					if (last != '.')
						ss << cnt << last;
					last = c;
					cnt = 1;
				} else
				{
					cnt++;
				}
			}
			ss << cnt << last;
			s = ss.str();
		}
		return s;
	}
};

File: 38.count-and-say/solution.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return '1'
        n = self.countAndSay(n - 1)
        last = '0'
        s = str(n)
        cnt = 0
        result = ''
        for i in range(len(s)):
            if s[i] == last:
                cnt += 1
            else:
                if cnt > 0:
                    result += str(cnt) + last
                cnt = 1
                last = s[i]
        return result + str(cnt) + last
